### ISSUE TITLE
Use rust-analyzer as default LSP server for Rust

### ISSUE THREAD

**Issue by Foltik** [doomemacs/doomemacs#6363]
### Describe your request

When the `+lsp` flag in the `rust` module is enabled, a language server isn't configured, so the default of RLS is used. `lsp-rust-server` should be explicitly set to `'rust-analyzer` instead.

### Briefly explain its use-case

It seems like the general consensus is that rust-analyzer provides a better editing experience out of the box than RLS, and the Rust community is moving towards adopting rust-analyzer as a standard.

See:
- https://github.com/rust-lang/rfcs/blob/master/text/2912-rust-analyzer.md
- https://old.reddit.com/r/rust/comments/lur37d/why_is_rls_still_so_bad/

**Comment by gagbo** [comment]
That's not true. The default, even on [Doom's pin](https://github.com/doomemacs/doomemacs/blob/258104f750fca0920850db8d6c9607ee7d81d846/modules/tools/lsp/packages.el#L9), is [rust-analyzer](https://github.com/emacs-lsp/lsp-mode/blob/1e7128c43f611ed78752130ce5ce67e641a276df/clients/lsp-rust.el#L49)

**Comment by Foltik** [comment]
Ah, you're right. I think my confusion came from the fact that:

- Rustup puts rls on the path even if the component is not installed (to print a message telling you to install it), so it's always picked up as a valid server by lsp-mode
- If rust-analyzer is missing from the path, lsp-mode automatically falls back to rls

This means that if neither language server is in the path, the user will get an error about how rls failed to start, with the error buffer containing a message saying to install rls.

I read this and figured I had to set `lsp-rust-server`, but even with an explicit value of `'rust-analyzer`, lsp-mode silently falls back to rls and displays the same error. I only figured out why when I looked through the logs and realized I didn't have rust-analyzer in my path.

I would expect there to be some sort of minibuffer message saying the configured lsp server is missing, as it wasn't clear whether it would be automatically downloaded or not.

If this is more of an issue for lsp-mode itself, this can be closed.
